src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (let i = 0; i < items.length; i++) {
+    const keyValue = items[i][key];
```

Comment:
The `keyValue` is used as an index for the `result` object, which assumes that `keyValue` is a string. However, `keyof T` could potentially be a non-string type. Consider adding a type assertion or conversion to ensure `keyValue` is a string.